src/plans/create/steps/GeneralInformationStep.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { FC, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { useNamespaces as useProviderNamespaces } from 'src/modules/Providers/hooks/useNamespaces';
+import ProviderSelect from 'src/plans/components/ProviderSelect';
+
+import ControlledFormGroup from '@components/common/ControlledFormGroup';
+import { useProjectNameSelectOptions } from '@components/common/ProjectNameSelect';
+import { TypeaheadSelect } from '@components/common/TypeaheadSelect';
+import { WizardStepContainer } from '@components/common/WizardStepContainer';
+import { V1beta1Provider } from '@kubev2v/types';
+import { useActiveNamespace } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Form,
+  FormSection,
+  MenuToggleStatus,
+  TextInput,
+  ValidatedOptions,
+} from '@patternfly/react-core';
+import { Namespace } from '@utils/constants';
+import { useForkliftTranslation } from '@utils/i18n';
+import { getDefaultNamespace } from '@utils/namespaces';
+
+import { GeneralFormFieldId, generalFormFieldLabels } from './constants';
+
+export const GeneralInformationStep: FC = () => {
+  const { t } = useForkliftTranslation();
+  const {
+    formState: { errors },
+    watch,
+    setValue,
+  } = useFormContext();
+  const [providerNamespace, setProviderNamespace] = useState<string>();
+  const planProject: string = watch(GeneralFormFieldId.PlanProject);
+  const targetProject: string = watch(GeneralFormFieldId.TargetProject);
+  const targetProvider: V1beta1Provider = watch(GeneralFormFieldId.TargetProvider);
+  const sourceProvider: V1beta1Provider = watch(GeneralFormFieldId.SourceProvider);
+  const [activeNamespace] = useActiveNamespace();
+  const defaultNamespace = getDefaultNamespace();
+  const initialProjectName =
+    planProject || (activeNamespace === Namespace.AllProjects ? defaultNamespace : activeNamespace);
+  const projectOptions = useProjectNameSelectOptions();
+  const [targetProviderProjects, targetProviderProjectsLoading] =
+    useProviderNamespaces(targetProvider);
+
+  return (
+    <WizardStepContainer title={t('General')}>
+      <Form>
+        <FormSection title={t('Plan information')} titleElement="h3">
+          <p>{t('Name your plan and choose the project you would like it to be created in.')}</p>
+
+          <ControlledFormGroup
+            isRequired
+            fieldId={GeneralFormFieldId.PlanName}
+            label={generalFormFieldLabels[GeneralFormFieldId.PlanName]}
+            controller={{
+              rules: { required: t('Plan name is required.') },
+              render: ({ field }) => (
+                <TextInput
+                  {...field}
+                  validated={
+                    errors[GeneralFormFieldId.PlanName]
+                      ? ValidatedOptions.error
+                      : ValidatedOptions.default
```

Comment:
Probably, this will be used a lot. Create a small shared function for it?

Reply:
I usually like to wait till there's another use-case to break it out into a shared helper, but most likely you're correct so I'll make one.